packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanDetail/KeyValuesTable.tsx

```diff
@@ -79,7 +71,26 @@ function formatValue(key: string, value: any) {
   } else if (Array.isArray(parsed) && shouldDisplayAsStringList(key)) {
     content = stringListMarkup(parsed);
   } else {
-    content = _jsonMarkup(parsed);
+    content = (
+      <JsonView
+        data={parsed}
+        shouldInitiallyExpand={allExpanded}
```

Comment:
could this be conditional on the size of the output, e.g. if the JSON object has more than 10 keys then don't start in expanded form.

Also, please post a screenshot of how it looks in the collapsed form.

Reply:
Here's how it looks when collapsed at the root

<img width="1333" alt="Screenshot 2023-08-24 at 12 55 08 PM" src="https://github.com/jaegertracing/jaeger-ui/assets/28570857/9b004ab2-f342-4d94-9583-b7ffd19ce21a">

Reply:
The JSON tree will now only expand if it has 10 or less keys, nested objects will remain collapsed otherwise